### ISSUE TITLE
time column in imu.data

### ISSUE THREAD

**Issue by philippcla** [SMAC-Group/gmwm#182]
Hello,
by using the "imu.data = read.imu(file = imu.file_path , type = imu.type);"
the data is assembled in 6 collums being Gyro.X, Gyro.Y, Gyro.Z, Accel.X, Accel.Y and Accel.Z. That is ok. BUT there is no column with the time data. it should look like the following
Time [s]** - GX - GY - GZ - AX - AY - AZ 

so it would be 7 columns

**Comment by coatless** [comment]
Work around for the moment is to use read_imu() note the **_** which gives direct access to the cop function.

**Comment by coatless** [comment]
Okay, would something along the lines of embedding the time information in the *rowname* of the matrix be acceptable?

e.g.

```r
a = matrix(c(1,0,0,1), nrow = 4)

# No names
a

rownames(a) = c(.3,.5,.6,.7)

# Named
a

times = rownames(a)
```

**Comment by coatless** [comment]
@philippcla Thoughts on the above? Would the proposed change be okay on your end?

**Comment by philippcla** [comment]
Yes

Once I have the indexes of the desired timespan (e.g. indexes 456 through
9857643, because I want to treat only data between two specific timestamps)
can I apply it on the IMU dataset?

original_imu_data = read.imu(file = imu.file_path , type = imu.type)
cutted_imu_data = original_imu_data[*indexes*,]

I keep the original, and will create a truncated dataset thanks to the*
indexes-vector*...
Is this possible?


2016-05-26 22:22 GMT+02:00 James J Balamuta <notifications@github.com>:

**Comment by coatless** [comment]
@philippcla Yes. You can subset under the current `imu` structure exactly like a matrix.

```r
original_imu_data = read.imu(file = imu.file_path , type = imu.type)
cutted_imu_data = original_imu_data[456:9857643,]
```

**Comment by coatless** [comment]
Per ddd7327, we have:

```r
original_imu_data = read.imu(file = imu.file_path , type = imu.type)
imu_time(original_imu_data)
```

(Might need to tweak print output to avoid renaming rownames...)
